### PR TITLE
Gamified merge final

### PR DESCRIPTION
**Pull request by loui981c**
## Description

Briefly describe the purpose and context of this pull request.

This pull request includes the implementation of the following PBIs:

- [App Login] As non-logged in user, I want to have the ability to preview the app, so that I can understand the application before register/login
- [App Home] As a waste picker, I want to view when I don't have an active course (empty state)
- [App Profile] As a waste picker, I would like to be able to see and edit my profile, in order to see what personal information is connected to my profile and edit in case something is incorrect or not up to date
- [App Profile] As a waste-picker, I want to have the ability to delete my account

## Changes

- Delete user implemented
- Profile settings page created
- Implemented ability to update first name, last name and email
- Added a welcome screen
- Updated screen when the user has no active courses
- Unit tests made
- Cleanup in all touched files
- Updated some of the routes in api.js and userApi.js

### Other changes 
- Snapshot fixes by using jest --watch
- React and Text imports

## Related Issues

Deletion of user can end up not working, as it hasn’t been tested with a logged in user. The same applies to update of first name, last name and email.

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)
Simulator_Screenshot_-_iPhone_15_Pro_-_2023-10-06_at_14 50 56
Simulator_Screenshot_-_iPhone_15_Pro_-_2023-10-06_at_14 58 28
Simulator_Screenshot_-_iPhone_15_Pro_-_2023-10-06_at_14 55 48

Profile settings page

## Notes for Reviewers
There are minor changes in most of the files

### CODE REVIEW

File: eml/.gitignore [hunk 1]
```diff
@@ -9,6 +9,8 @@ npm-debug.*
 *.orig.*
 web-build/
 eslint-demo
+package-lock.json
+package.json
```
Comment (Vagn99) [inline eml/.gitignore:13]: Er I sikre på package.json skal med i .gitignore?? Så kan andre jo ikke installere de packages der skal bruges😁

File: eml/api/api.js [hunk 2]
```diff
@@ -25,29 +25,29 @@ export const getAuthToken = async () => {
 
 export const getTestCourse = async () => {
   const res = await axios.get(
-    url + '/api/public/courses/635fb5b9b2fb6c4f49084682'
+    url + '/api/public/course/635fb5b9b2fb6c4f49084682'
```
Comment (Vagn99) [inline eml/api/api.js:28]: Faktisk kommer ruten til at være ".../courses/:courseId" og ikke ".../course/:courseId". Ved dog ikke om det skal ændres allerede nu i appen 🤔

**Review by Vagn99 — COMMENTED** [review]
Har lige to kommentarer😁💯💸

**Review by Vagn99 — APPROVED** [review]
Very good, very nice

**Comment by Vagn99** [comment]
Brug gerne "Squash and merge" således at det kun er "ét" commit der kommer ind :) @Havkost

File: eml/components/profileSettings/changeEmailButton.js
Comment (Havkost) [inline eml/components/profileSettings/changeEmailButton.js:1]: Ville være fedt at få testet change email.

File: eml/components/profileSettings/changeFirstNameButton.js
Comment (Havkost) [inline eml/components/profileSettings/changeFirstNameButton.js:1]: Ville være fedt at få testet change first name.

File: eml/components/profileSettings/changeLastNameButton.js
Comment (Havkost) [inline eml/components/profileSettings/changeLastNameButton.js:1]: Ville være fedt at få testet change last name.

File: eml/components/profileSettings/deleteAccountButton.js
Comment (Havkost) [inline eml/components/profileSettings/deleteAccountButton.js:1]: Ville være fedt at få testet delete account.

**Review by Havkost — APPROVED** [review]
## Approved
Godt arbejde.

Blokerer ikke merget nu pga. det, men jeg synes der er meget få tests. Specielt funktionaliteter som dem jeg har kommenteret på ville være fedt at testet fremover. (med mindre jeg bare har misset dem)

Jeg kan ikke få koden til at køre lige nu, men jeg har på fornemmelsen det bare er min emulator.

Husk at kommetere en engelsk version når I bruger portugisiske strenge.

Ser ikke ud som om I har brugt den merge metode Toby viste den anden dag? Ville være nice at gøre det fremover, så ikke vi skal review de samme filer igen og igen.

**Comment by sofiiagran** [comment]
problems running github actions?

**Pull request merged**